wemake_python_styleguide/visitors/ast/complexity/function.py

```diff
@@ -20,27 +21,31 @@
 from wemake_python_styleguide.visitors.base import BaseNodeVisitor
 from wemake_python_styleguide.visitors.decorators import alias
 
-_FunctionCounter = DefaultDict[AnyFunctionDef, int]
-_FunctionCounterWithLambda = DefaultDict[AnyFunctionDefAndLambda, int]
-_AnyFunctionCounter = Union[_FunctionCounter, _FunctionCounterWithLambda]
-_CheckRule = Tuple[_AnyFunctionCounter, int, Type[BaseViolation]]
+_FunCt = DefaultDict[AnyFunctionDef, int]
+_FunCtWithLambda = DefaultDict[AnyFunctionDefAndLambda, int]
+_FunCtVars = DefaultDict[AnyFunctionDef, List[str]]
+_AnyFunCt = Union[_FunCt, _FunCtWithLambda]
+_CheckRule = Tuple[_AnyFunCt, int, Type[BaseViolation]]
 _NodeTypeHandler = Dict[
     Union[type, Tuple[type, ...]],
-    _FunctionCounter,
+    _FunCt,
 ]
 
 
 @final
-class _ComplexityExitMetrics(object):
+@attr.dataclass(frozen=False)
+class _ComplexityMetrics(object):
     """
     Helper class.
 
-    Stores counters of statements that exit from a function.
+    Stores counters of function internals.
     """
 
-    def __init__(self) -> None:
-        self.returns: _FunctionCounter = defaultdict(int)
-        self.raises: _FunctionCounter = defaultdict(int)
+    returns: _FunCt = attr.ib(factory=lambda: defaultdict(int))
```

Comment:
`lambda: defaultdict(int)` can be turned into a new function. Like:

```python
def _counter_factory() -> DefaultDict[AnyFunctionDef, int]:
    return defaultdict(int)
```

And then use it!